### PR TITLE
Restructure and add description for event types.

### LINKED ISSUE #297: Official Definitions for Transaction and Trace
Request for official definitions for these in the docs. There seems to be some variation depending on the agent. This is more useful for messaging and when communicating in the field.

@asawariS @jamiesmith

### PR DESCRIPTION
**Pull request by simitt**
Define transaction, span and error and restructure documentation.
closes #297

I seperated the description of the events and the API, as the API will mainly only interest agent developers, whereas the general description of events is interesting for all customers.

 
screen shot 2018-01-26 at 10 43 36

screen shot 2018-01-26 at 10 43 43

### CODE REVIEW

File: docs/context.asciidoc [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+the `system` in which the monitored service is running and the event's `process` information.
+
+It can also contain information about the authenticated `user`. 
+
+An event's context can also include information about the request leading to the event and the response of processing the event. 
+In case a http request is captured, it contains information about the `url`, `cookies`, `body`, `headers`, etc.
+
+The agents provide some configuration options with which the users can also capture customized information.
+The non-indexed information is captured within a `custom` object,
+while the indexed information is stored within `tags`.
```
Comment (graphaelli) [inline docs/context.asciidoc]: are we good to assume users know what indexed means?  maybe searchable instead ?

File: docs/errors.asciidoc [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+include::./context.asciidoc[]
+
+[[errors-error]]
+[float]
+==== Error
+An error event contains at least
+information about the original `exception` that occured 
+or information about a `log` that was created when the exception occured. 
+
+Both, the captured `exception` and the captured `log` of an error can contain `stack trace` information,
```
Comment (graphaelli) [inline docs/errors.asciidoc]: stray comma after `Both`

File: docs/spans.asciidoc [hunk 3]
```diff
@@ -0,0 +1,43 @@
+[[spans]]
+=== Spans
+
+A span contains information about a specific code path, 
+executed as part of a <<transactions,transaction>>. 
+Every span is identified by an ID unique per transaction.
```
Comment (graphaelli) [inline docs/spans.asciidoc]: `s/an ID unique/a unique ID/`

**Comment by simitt** [comment]
Thank you @graphaelli for the review, I also think we should update the term `frontend` later on, once it is fixed.

**Pull request merged**